src/Portability/Driver.php

```
@@ -73,4 +72,22 @@ public function connect(
             new Converter($convertEmptyStringToNull, $rightTrimString, $case),
         );
     }
+
+    /**
+     * @param 0|ColumnCase::LOWER|ColumnCase::UPPER $case
+     *
+     * @return PDO::CASE_LOWER|PDO::CASE_UPPER|PDO::CASE_NATURAL
+     */
+    private static function convertToPdoCase(int $case): int
+    {
+        if ($case === ColumnCase::LOWER) {
+            return PDO::CASE_LOWER;
+        }
+
+        if ($case === ColumnCase::UPPER) {
+            return PDO::CASE_UPPER;
+        }
+
+        return PDO::CASE_NATURAL;
```

This method is never called with `0` so this `return` is unreachable code. Let's remove it.

---

Should we throw an exception here?

---

I don't think we need to. Make this method a ternary expression and inline it. That would also be consistent with the code for non-PDO connections in line 59.

```php
if ($nativeConnection instanceof PDO) {
    $portability &= ~Connection::PORTABILITY_FIX_CASE;
    $nativeConnection->setAttribute(PDO::ATTR_CASE, ColumnCase::LOWER ? PDO::CASE_LOWER : PDO::CASE_UPPER);
} else {
    $case = $this->case === ColumnCase::LOWER ? Converter::CASE_LOWER : Converter::CASE_UPPER;
}
```

---

Done.